addons/purchase/models/purchase.py

```diff
@@ -1106,6 +1106,7 @@ def _onchange_quantity(self):
             price_unit = seller.product_uom._compute_price(price_unit, self.product_uom)
 
         self.price_unit = price_unit
+        self.name = self._get_product_purchase_description(self.product_id.with_context(seller_id=seller.id))
```

Comment:
I'm not sure if it matters, but `_get_product_purchase_description` seems to also expect the partner's language in the context, i.e. maybe we should make it consistent? with https://github.com/odoo/odoo/blob/a0d55d5971cfcfa1279bbbd6a7ee926f9dfea34b/addons/purchase/models/purchase.py#L1036-L1041